tsconfig.json

```diff
@@ -0,0 +1,46 @@
+{
+  "compileOnSave": false,
+  "compilerOptions": {
+    "allowUnreachableCode": false,
+    "declaration": false,
+    "downlevelIteration": true,
+    "importHelpers": true,
+    "inlineSourceMap": true,
+    "inlineSources": true,
+    "module": "esnext",
+    "moduleResolution": "node",
+    "noEmitOnError": true,
+    "noErrorTruncation": true,
+    "allowSyntheticDefaultImports": true,
+    // "noFallthroughCasesInSwitch": true,
+    // "noImplicitAny": true,
+    "noImplicitReturns": true,
+    "noImplicitThis": true,
+    "strict": false,
+    "lib": [
+      "es5",
+      "dom",
+      "es2015.promise",
+      "es2015.symbol",
+      "es2015.iterable",
+      "es2015.collection",
+      "es2015.symbol.wellknown",
+      "es2015.core",
+      "ES2016.Array.Include",
+      "ES2017.Intl",
+      "ES2017.Object",
+      "ES2017.String",
+    ],
+    "preserveConstEnums": false,
+    "skipDefaultLibCheck": true,
+    "sourceMap": false,
+    // "strictBindCallApply": true,
+    // "strictFunctionTypes": true,
+    // "strictNullChecks": true,
+    // "strictPropertyInitialization": true,
+    "stripInternal": true,
+    "target": "es5",
```

Comment:
This is probably my unfamiliarity with TypeScript configuration, but how is it outputting ESM files for an ES5 target?

Reply:
https://github.com/Microsoft/TypeScript/issues/6319
https://github.com/microsoft/TypeScript/issues/5790

These imply to me that the options here are asking tsc to emit ES5 code using the ES6 module format instead of defaulting to commonjs. It seems the module format is only loosely tied to the language output level.

I too don't know for sure if this is "right" - I was basing it off the internal tsconfig used at Google but with modifications to get the existing test suite to run (which needed ESM). If we were to change a value here, I think it would be to change target to something higher and let the rest of the rollup config handle that output.

Reply:
That being said, I did adjust the module output to explicitly be es6 instead of the unstable esnext.